### PR TITLE
Aaron/addspawningtests

### PR DESCRIPTION
**Pull request by Blyron**
Adding Carla integration basic Tests. 
Moving Beginplay out of commented scope due to BeginPlay is need to make actors spawn properly

### CODE REVIEW

File: PythonAPI/test/API/collision_test.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    pass
+
+import carla
+
+
+#record the result as json to the same folder
+
+
+class TestVehiclesSpawnTest(unittest.TestCase):
+    collisions = 0
```
Comment (xavisolesoft) [inline PythonAPI/test/API/collision_test.py:30]: The correct way to define class members in python is like this: `self.collisions`.

Concretely, in python unittest, this is the correct why:
```
def setUp(self):
    self.collisions = 0
```
And refer collisions always as `self.collisions = 0`.

File: PythonAPI/test/API/spawn_vehicles.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    pass
+try:
+    sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + '/carla')
+except IndexError:
+    pass
+
+import carla
+
+class TestVehiclesSpawnTest(unittest.TestCase):
+    print("IntegrationTest.Vehicles.VehiclesSpawnTest")
```
Comment (xavisolesoft) [inline PythonAPI/test/API/spawn_vehicles.py:22]:
```suggestion
    print("IntegrationTest.Vehicles.VehiclesSpawnTest")
```
This kind of prints are not needed, so rest running engines like pytest add this kind of logs.

File: PythonAPI/test/integration_tests_list.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+API.spawn_walkers API.spawn_vehicles
```
Comment (xavisolesoft) [inline PythonAPI/test/integration_tests_list.txt:1]: Should we add also collision test here?

File: PythonAPI/test/API/collision_test.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+except IndexError:
+    pass
+
+import carla
+
+
+#record the result as json to the same folder
+
+
+class TestVehiclesSpawnTest(unittest.TestCase):
```
Comment (xavisolesoft) [inline PythonAPI/test/API/collision_test.py:29]:
```suggestion
class TestCollision(unittest.TestCase):
```

File: PythonAPI/test/API/spawn_vehicles.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+except IndexError:
+    pass
+try:
+    sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + '/carla')
+except IndexError:
+    pass
+
+import carla
+
+class TestVehiclesSpawnTest(unittest.TestCase):
```
Comment (xavisolesoft) [inline PythonAPI/test/API/spawn_vehicles.py:21]:
```suggestion
class TestVehiclesSpawn(unittest.TestCase):
```

Name simplification

File: PythonAPI/test/API/spawn_walkers.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+except IndexError:
+    pass
+try:
+    sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + '/carla')
+except IndexError:
+    pass
+
+import carla
+
+class TestWalkersSpawnTest(unittest.TestCase):
```
Comment (xavisolesoft) [inline PythonAPI/test/API/spawn_walkers.py:21]:
```suggestion
class TestWalkersSpawn(unittest.TestCase):
```
Name simplification

File: PythonAPI/test/integration_tests_list.txt (hunk 3 above)
Comment (Blyron) [reply PythonAPI/test/integration_tests_list.txt:1]: Not yet as vehicles are not working in UE5. I want to keep the record of the file to add sameones to UE4

**Review by xavisolesoft — CHANGES_REQUESTED** [review]

**Review by xavisolesoft — APPROVED** [review]

**Pull request merged**